src/subcommands/list_opcodes.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Lists the supported opcodes for a given provider.
+
+use crate::cli::ParsecToolApp;
+use crate::error::ParsecToolError;
+use crate::subcommands::common::ProviderOpts;
+use crate::subcommands::ParsecToolSubcommand;
+use parsec_client::core::interface::requests::ProviderID;
+use parsec_client::core::operation_client::OperationClient;
+use parsec_interface::operations::list_opcodes;
+use parsec_interface::operations::{NativeOperation, NativeResult};
+use std::convert::TryFrom;
+use structopt::StructOpt;
+
+/// Lists the supported opcodes for a given provider.
+#[derive(Copy, Clone, Debug, StructOpt)]
+#[structopt(name = "list_opcodes")]
+pub struct ListOpcodesSubcommand {
+    #[structopt(flatten)]
+    provider_opts: ProviderOpts,
+}
+
+impl TryFrom<ListOpcodesSubcommand> for NativeOperation {
+    type Error = ParsecToolError;
+
+    fn try_from(list_opcodes_subcommand: ListOpcodesSubcommand) -> Result<Self, Self::Error> {
+        // Trivially converted to a `NativeOperation`.
+        Ok(NativeOperation::ListOpcodes(list_opcodes::Operation {
+            provider_id: ProviderID::try_from(list_opcodes_subcommand.provider_opts.provider)?,
+        }))
+    }
+}
```

Comment:
Nice! It's a very good idea to use the `OperationClient` here. Also each command will make its own choice of either using the basic client or the operation client 👌